http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -262,8 +262,7 @@ T get()
         private final LoginContext loginContext;
         private final GSSCredential clientCredential;
 
-        public Session(LoginContext loginContext, GSSCredential clientCredential)
```

Comment:
I disagree with this change. While this is valid under the Java language rules for nested classes (even private would work), the original modifiers better convey the logical intent of the methods from an API standpoint. They are public because they are part of the public API of the class. Marking them private conveys that they are for use inside the class.

One way to think about this is to consider what happens if we need to increase the visibility of the class to be package private or public. (Similarly, the same argument applies to increasing the visibility of a package private class to be public.) If we use public/private to convey the intent of each API member, then this change only requires changing the modifier of the class. If we went with what is allowed by the Java language (anything goes), then we would need to go through each method and consider the visibility and if it should be part of the public API.